include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -337,18 +337,28 @@ namespace internal
           AssertIndexRange(0, n_hp_quads);
           cell_data[my_q].descriptor.resize(n_hp_quads);
           for (unsigned int q = 0; q < n_hp_quads; ++q)
-            cell_data[my_q].descriptor[q].initialize(quad[my_q][q],
-                                                     update_default);
+            {
+              Assert(quad[my_q][q].is_tensor_product(), ExcNotImplemented());
+              cell_data[my_q].descriptor[q].initialize(
+                quad[my_q][q].get_tensor_basis()[0], update_default);
```

Comment:
In principle, we would also need a second assert here that checks that all of the 1D quadrature formulas returned by `get_tensor_basis` are the same. While algorithms could be extended, that would be a lot of work and we should catch it before getting unexpected results.